src/components/Demo/DemoCreate/DemoCreateForm/DemoCreateFormItem/index.tsx

```diff
@@ -0,0 +1,76 @@
+import { Input } from "@govtechsg/tradetrust-ui-components";
+import React from "react";
+import { FunctionComponent } from "react";
+import { AccordionItem } from "../../../../UI/Accordion";
+import { getFormValue } from "../../utils";
+import { data } from "../data";
+import { FormItemSchema } from "../types";
+
+interface DemoCreateFormItemProps {
+  onChange: (name: string, value: string) => void;
+  formItem: FormItemSchema;
+  formItemName: string;
+}
+
+export const DemoCreateFormItem: FunctionComponent<DemoCreateFormItemProps> = ({
+  formItem,
+  formItemName,
+  onChange,
+}) => {
+  const renderProperties = () => {
+    if (formItem.type === "object" && formItem.properties) {
+      return Object.entries(formItem.properties).map(([name, item]) => {
+        //join by . delimiter to parse later
+        const itemName = `${formItemName}.${name}`;
+
+        return <DemoCreateFormItem key={name} onChange={onChange} formItemName={itemName} formItem={item} />;
```

Comment:
curious why decide to construct the component in a recursive manner instead of building it up from bottom?

Reply:
less code to write and also accomodate for possible further nesting of objects

Reply:
have you thought of the complexity that might come up in the future if we have other logic branches?

Reply:
yes but considering this is component is for the demo only I figured it wouldn't be much more complex

Reply:
okay sure 👌

Reply:
@cavacado let me know if you need me to change this otherwise ill merge it in